token/js/src/extensions/interestBearingMint/instructions.ts

```diff
@@ -0,0 +1,96 @@
+import { struct, s16, u8 } from '@solana/buffer-layout';
+import { publicKey } from '@solana/buffer-layout-utils';
+import { PublicKey, TransactionInstruction } from '@solana/web3.js';
+import { TOKEN_2022_PROGRAM_ID } from '../../constants';
+import { TokenInstruction } from '../../instructions';
+
+export enum InterestBearingMintInstruction {
+    Initialize = 0,
+    UpdateRate = 1,
+}
+
+export interface InterestBearingMintInitializeInstructionData {
+    instruction: TokenInstruction.InterestBearingMintExtension;
+    interestBearingMintInstruction: InterestBearingMintInstruction.Initialize;
+    rateAuthority: PublicKey;
+    rate: number;
+}
+
+export interface InterestBearingMintUpdateRateInstructionData {
+    instruction: TokenInstruction.InterestBearingMintExtension;
+    interestBearingMintInstruction: InterestBearingMintInstruction.UpdateRate;
+    rate: number;
+}
+
+export const interestBearingMintInitializeInstructionData = struct<InterestBearingMintInitializeInstructionData>([
+    u8('instruction'),
+    u8('interestBearingMintInstruction'),
+    publicKey('rateAuthority'),
```

Comment:
I'm just realizing that this isn't done correctly with the transfer fee extension, since it's not actually a `publicKey`, but rather an `OptionalNonZeroPublicKey`, meaning that it's really `publicKey?`.  We can keep this for now, but eventually we need to properly handle the optionality

Reply:
gotcha. is that type typicall added to `@solana/buffer-layout-utils` or are they usually defined in the token js library?

Reply:
In this case, we can probably add it to the token js library.  If it eventually gets ported to the solana sdk, then we can also add it to buffer-layout-utils